[15.0][MIG] base_delivery_carrier_label: Migration to 15.0

fshah-initos opened the pull request:
Dependent module: delivery_carrier_info 
Pr: https://github.com/OCA/delivery-carrier/pull/475

yvaucher committed:
[ADD] base_delivery_carrier_label portage of Akretion delivery_shipping_label

Commit:
[CHG] pep-8 and pep-257, return eagerly in option_ids_change to avoid nesting, few typos

[ADD] access rights for the shipping.label model

yvaucher committed:
[CHG] contributors

[IMP] remove unused pdf_list variable and rename pdf variable to label as a label can have other printing format

[IMP] remove readonly on carrier_tracking_ref

[IMP] remove carrier_tracking_ref redefinition as already defined in delivery/stock.py

[ADD] stock users rights

[ADD] file type on shipping label

[ADD] pot file

[ADD] fr.po

jgrandguillaume committed:
[DEL] Remove unused views

yvaucher committed:
[IMP] add tracking_id on shipping label to be able to create a label per pack (stock.tracking)

[IMP] set tracking_id as an optional key of dict returned by label generation and improve doc string

[IMP] rename generate_pack_labels in generate_shipping_labels as it can also generate a single label for a picking without pack notion

[IMP] set labels on stock.picking instead of stock.picking.out in order to let the creation of return labels to provide to the customer

[FIX] xml view and copy what is needed on stock.picking.out and stock.picking.in as it won't be inherited from stock.picking.out

Commit:
[FIX] load the ir.model.access.csv file

[FIX] delivery.carrier has no access rights for stock users, hence remove it for delivery.carrier.option

yvaucher committed:
[FIX] Replace NotImplemented exception which was wrongly interpreted by raising a orm.except_orm

[FIX] trigger carrier_id on change when creating a stock.picking to set default options from sale.order

[IMP] change error message of unimplemented default to a more user friendly message

bealdav committed:
[IMP] add a menu entry 'Carriers' in settings

Commit:
[ADD] an optional parameter 'tracking_ids' so we can decide to generate the label only for a selection of packs in a picking

the button call action_generate_carrier_label() without the 'context' keyword argument so the context ends in the tracking_ids argument. So the method now conforms to the cr, uid, ids, context signature and just forwards to a new generate_labels() method that should be used from within the code

do not overwrite the option_ids on create of stock.picking if already set in values, generalize the method to all types of picking (added in/out)

Set the default options of the delivery method when it is changed and the options are not set (typically a write from the code or a mass update)

yvaucher committed:
base_delivery_carrier_label - Improve carrier type selection inheritance

docstring

Commit:
Launchpad automatic translations update.

bealdav committed:
[IMP] add common view action to use by  module install

[FIX] action name

[IMP] add descrption field to carrier option

yvaucher committed:
When tracking_ids is defined, generate one default label per tracking id

bealdav committed:
[IMP] complete translation and fix view

[IMP] file_type field inheritance in shipping.label

[IMP] add field selection sort

Commit:
Launchpad automatic translations update.

bealdav committed:
[IMP] add _customize_sender_address method

[FIX] docstring

[FIX] _customize_sender_address return method

[IMP] add comment in _customize_sender_adress()

[FIX] method name to get sender address

[FIX] method comment

yvaucher committed:
set installable to false

move modules in __unported__

bealdav committed:
[REF] remove stock.picking.in and .out

[REF] port to v8 with old api

[MIG] to v8 base_deliv_label, deliv_carrier_deposit

[IMP] add parcel_tracking field

[IMP] add _get_packages_from_picking()

add carrier_id in tree view

alexis-via committed:
[FIX] add explicitely attachment_id (previously included by orm)

[FIX] complete view

bealdav committed:
[FIX] weight field type

Commit:
Add weight in form view of stock.quant.package
Better layout of default report for deposit slip
For stock.quant.package, order by id desc to have the latest first
Code cleanup

Button "Create Shipping Label" should be used when the picking in Done, because in v8 the packaging is handled in the "Transfer" wizard (in v7, the packaging could be made when the picking is waiting/ready)
Display number of packages on deposit slip
Add weight in tree view of packages

Commit:
Migrate to new API

Remove an unused file 'reports.xml'

Remove the selection wrappers

Use the name of the method as string instead

Commit:
Remove _order on stock.quant.packages, as asked by Guewen

bealdav committed:
[FIX] writing package_id field in shipping.label

rschnapka committed:
add and generate initial german translations

Commit:
[Usability] Better name_get on stock.quant.package

Move module description in README.rst

Commit:
Allow to generate labels when we moved an entire package

Take the destination package. If empty, the package is moved so take the source
one.

The options must be filtered by the selected carrier

cubells committed:
[NEW] Added spanish translation

gurneyalex committed:
Add OCA as author of OCA addons

In order to get visibility on https://www.odoo.com/apps the OCA board has
decided to add the OCA as author of all the addons maintained as part of the
association.

yvaucher committed:
[PEP8]

Commit:
Do not return duplicate packages when several moves have the same one

Because it will print several labels for the same package.

sbidoul committed:
[UPD] prefix versions with 8.0

yvaucher committed:
[PORT] base_delivery_carrier_label port to 9.0
- Reactivate
- Move files in views and models
- Adapt XML views
- Update README.rst

Move settings menu in Inventory/Configuration

Remove test with Logistic units as it was removed from 9.0

Allow to give a filename to the created label if not defined use the attachment name

Fix weight calcul adding a field for total weight
This due to removed logistics units
If set this field will be used to define weight, otherwise we compute
weight even before quants have been created.

Fix stock picking view some labels were missing and the view was scrambled

Split files by model

Commit:
ADD manifest and MIG 10

Commit:
[IMP] add get_weight() to quant package

Commit:
[IMP] put set_pack_weight() in picking

bealdav committed:
[IMP] put _check_existing_shipping_label() in picking

[FIX] oca remarks

hparfr committed:
refactore stock.get_weight()

add operation.get_weight()
add tests

fix wrong name of class

refactor tests for a better readability

s/tracking/package in docstring of generate_labels

Commit:
Rename field type to carrier_type in delivery.carrier

Missing acl for stock user/manager

angelmoya committed:
WIP migraciont label

MIG 10.0 base_delivery_carrier_label

angelmoya committed:
FIX manifest wizard as transient model

IMP modularity and fixes

mdietrichc2c committed:
Use correct syntax to add options with onchange on stock picking

Commit:
[FIX] base_delivery_carrier_label: Fix security.
* Remove manifest.wizard model access rules.

lasley committed:
[IMP] base_delivery_carrier_label: Add quant package tracking URI
* Add an attribute for the carrier tracking URI at the `stock.quant.package` level

hugosantosred committed:
[MIG] 11.0 base_delivery_carrier_label

asaunier committed:
[IMP] Improve migration to 11.0

grindtildeath committed:
[FIX] Remove occurences of carrier type

Field carrier_type was originally type and was used to differentiate delivery carriers.
However odoo did implement delivery_type since v9 with this exact goal :
https://github.com/odoo/odoo/blob/11.0/addons/delivery/models/delivery_carrier.py#L22
This is also the field used in the implementation of enterprise modules for carriers.
Therefore we should use it instead of redefining a new one..

grindtildeath committed:
Add oldname to delivery_type

Timo17100-c2c committed:
[MIG] base_delivery_carrier_label: Migration to 12.0

bealdav committed:
REFactor base_deliv_carr_label: always use package

bealdav committed:
IMP add view to carrier.account

sbidoul committed:
Remove duplicate entry in fa.po

florian-dacosta committed:
[FIX] Wait data encoded in base64 / write tracking number on package

liweijie0812 committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 100.0% (96 of 96 strings)

Translation: delivery-carrier-12.0/delivery-carrier-12.0-base_delivery_carrier_label
Translate-URL: https://translation.odoo-community.org/projects/delivery-carrier-12-0/delivery-carrier-12-0-base_delivery_carrier_label/zh_CN/

florian-dacosta committed:
[IMP] Add delivery type on carrier acount to be able to link easily an account to a type of carrier

ps-tubtim committed:
[IMP] base_delivery_carrier_label: black, isort

ps-tubtim committed:
[MIG] base_delivery_carier_label: Migration to 13.0

sebastienbeau committed:
extract delivery_carrier_info from base_delivery_carrier_label

bealdav committed:
FIX base_deliv_label: use valid qty to compute weight of move.line

hbrunn committed:
[ADD] active flag for options

hbrunn committed:
[ADD] generic test case for labels

hbrunn committed:
[ADD] handle labels created during shipping if any

hbrunn committed:
[IMP] determine generate label button's visibility with a computed field

hbrunn committed:
[ADD] carrier.account#sequence

helper function to get an account for a picking

hbrunn committed:
[ADD] test helper functions

oca-travis committed:
[UPD] Update base_delivery_carrier_label.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: delivery-carrier-13.0/delivery-carrier-13.0-base_delivery_carrier_label
Translate-URL: https://translation.odoo-community.org/projects/delivery-carrier-13-0/delivery-carrier-13-0-base_delivery_carrier_label/

sebastienbeau committed:
[FIX] fix migration

datas_fname do not exist anymore in V13
clean super call
add missin open_website_url
clean mock testing

sebastienbeau committed:
[IMP] base_delivery_carrier_label: black, isort, prettier

sebastienbeau committed:
[MIG] migration to V14

florian-dacosta committed:
Fix common carrier label tests

florian-dacosta committed:
Refactore base_delivery_carrier_label to improve compatibility with native way to generate labels
- Remove action_generate_carrier_label button in favor on native send_to_shipper button
- Remove show_label_button field as we now have only one button to display
- Avoid generating label if a tracking ref is already set on picking
- remove total weight on packages as the native shipping weight should do the same job
- Add hook to be able to generate label with method not delivery type specific

florian-dacosta committed:
Fix test : show_label_button does not exist anymore

StefanRijnhart committed:
[RFR] label case: allow to override transfer in setup

StefanRijnhart committed:
[RFR] Extract Test classes from Case classes for easier overriding

florian-dacosta committed:
Fix docstrings

oca-travis committed:
[UPD] Update base_delivery_carrier_label.pot

OCA-git-bot committed:
[UPD] README.rst

florian-dacosta committed:
Allow to have multiple carrier accounts for a same delivery type and still be able to select the right account depending on delivery methods

StefanRijnhart committed:
[12.0][RFR] base_delivery_carrier_label: show options as tags

Mandatory tags show up as orange in the option widget on the picking.

astirpe committed:
[FIX] delivery.carrier.option form view

astirpe committed:
[FIX] delivery.carrier.template.option form view

oca-travis committed:
[UPD] Update base_delivery_carrier_label.pot

OCA-git-bot committed:
base_delivery_carrier_label 14.0.1.1.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: delivery-carrier-14.0/delivery-carrier-14.0-base_delivery_carrier_label
Translate-URL: https://translation.odoo-community.org/projects/delivery-carrier-14-0/delivery-carrier-14-0-base_delivery_carrier_label/

florian-dacosta committed:
Fix carrier account intercompany security rule

Commit:
[14.0] base_delivery_carrier_label: add check when no carrier_id assigned and fix import mock bug

hailangvn committed:
[IMP] base_delivery_carrier_label: allow bypassing default…

… package in _set_a_default_package.

Yvesldff committed:
Translated using Weblate (French)

Currently translated at 50.0% (46 of 92 strings)

Translation: delivery-carrier-14.0/delivery-carrier-14.0-base_delivery_carrier_label
Translate-URL: https://translation.odoo-community.org/projects/delivery-carrier-14-0/delivery-carrier-14-0-base_delivery_carrier_label/fr/

OCA-git-bot committed:
base_delivery_carrier_label 14.0.1.1.1

oca-travis committed:
[UPD] Update base_delivery_carrier_label.pot

OCA-git-bot committed:
base_delivery_carrier_label 14.0.1.2.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: delivery-carrier-14.0/delivery-carrier-14.0-base_delivery_carrier_label
Translate-URL: https://translation.odoo-community.org/projects/delivery-carrier-14-0/delivery-carrier-14-0-base_delivery_carrier_label/

OCA-git-bot committed:
[IMP] update dotfiles [ci skip]

OCA-git-bot committed:
base_delivery_carrier_label 14.0.1.2.1

fshah-initos commented:
Test case and runboat/build error were not solved because the dependent module is not merged in main branch 15.0.

florian-dacosta commented:
Hello, same problem regarding the commit history here, please follow the method described here : https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-15.0

rousseldenis commented:
/ocabot migration base_delivery_carrier_label

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 10.6% (10 of 94 strings)

Translation: delivery-carrier-14.0/delivery-carrier-14.0-base_delivery_carrier_label
Translate-URL: https://translation.odoo-community.org/projects/delivery-carrier-14-0/delivery-carrier-14-0-base_delivery_carrier_label/it/

primes2h committed:
Translated using Weblate (Italian)

Currently translated at 12.7% (12 of 94 strings)

Translation: delivery-carrier-14.0/delivery-carrier-14.0-base_delivery_carrier_label
Translate-URL: https://translation.odoo-community.org/projects/delivery-carrier-14-0/delivery-carrier-14-0-base_delivery_carrier_label/it/

grindtildeath committed:
base_delivery_carrier: Fix alternative_send_shipping return value

Function delivery.carrier.alternative_send_shipping is called by
the override of send_shipping (if the call to its super didn't return
any result) and its result is returned by send_shipping.
However, as send_shipping is called by stock.picking.send_to_shipper
in the delivery module with a slicing operator, a list of at least
one element is expected and returning an empty dictionary is raising
a KeyError.
Instead of returning an empty dictionary, the function
alternative_send_shipping must return a list containing a dictionary
with keys `exact_price` and `tracking_number` because these keys are
used to access the result's values inside send_to_shipper function in
the delivery module.

OCA-git-bot committed:
base_delivery_carrier_label 14.0.1.2.2

hailangvn committed:
[FIX] base_delivery_carrier_label: avoid duplicating carrier_id

dsolanki-initos reviewed (commented):
Code review and functional test.
Looks good to me and except depended module failed case

OCA-git-bot committed:
base_delivery_carrier_label 14.0.1.3.0

ikapasi-initos commented:
Hi, @sebalix can you please review this pull..?

ikapasi-initos commented:
Hi, @CarlosRoca13 can you please review this pull..?

CarlosRoca13 reviewed (approved):
Please separate the initial pre-commit changes from the migration commit.

sebalix reviewed (commented):
Hi, could you integrate last commits from 14.0 branch? There are missing commits for version 14.0.1.2.2 and 14.0.1.3.0 (see https://github.com/OCA/delivery-carrier/commits/14.0/base_delivery_carrier_label )

ikapasi-initos commented:
> Hi, could you integrate last commits from 14.0 branch? There are missing commits for version 14.0.1.2.2 and 14.0.1.3.0 (see https://github.com/OCA/delivery-carrier/commits/14.0/base_delivery_carrier_label )

Hi, @sebalix I've added missing commits for versions 14.0.1.2.2 and 14.0.1.3.0.
Can you please review this pull..?

ikapasi-initos commented:
@chienandalu can you please review this pull..?

chienandalu commented:
> @chienandalu can you please review this pull..?

I'm afraid we don't work with that family of modules. Better ask to the usual mantainers.

ikapasi-initos commented:
@ivantodorovich can you please review this pull..?

rousseldenis commented on base_delivery_carrier_label/models/delivery_carrier.py:
```diff
@@ -16,7 +16,7 @@ class DeliveryCarrier(models.Model):
 
     def alternative_send_shipping(self, pickings):
-        return [{"exact_price": False, "tracking_number": False}]
+        return {}
```
@fshah-initos Why removing this ?

ikapasi-initos replied:
@rousseldenis I've removed it for testing purposes now I added it.

dsolanki-initos commented:
Hello @gurneyalex, This PR is approved by @OCA-git-bot, Could you please review and merge this PR?

rousseldenis reviewed (approved):
Code review

dsolanki-initos committed:
[IMP] base_delivery_carrier_label: black, isort, prettier

dsolanki-initos committed:
[MIG] base_delivery_carrier_label: Migration to 15.0

jans23 commented:
@rousseldenis This PR has been approved. Anything we can do to get it merged?

rousseldenis commented:
/ocabot merge patch
